adapters/flipp/flipp.go

```diff
@@ -0,0 +1,199 @@
+package flipp
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/buger/jsonparser"
+	"github.com/gofrs/uuid"
+	"github.com/prebid/openrtb/v17/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+var BANNER_TYPE = "banner"
+var INLINE_DIV_NAME = "inline"
+var AD_TYPES = []int64{4309, 641}
+var DTX_TYPES = []int64{5061}
+var COUNT = int64(1)
+var FLIPP_BIDDER = "flipp"
+var DEFAULT_CURRENCY = "USD"
+
+type adapter struct {
+	endpoint string
+}
+
+// Builder builds a new instance of the Foo adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	bidder := &adapter{
+		endpoint: config.Endpoint,
+	}
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	adapterRequests := make([]*adapters.RequestData, 0, len(request.Imp))
+	var errors []error
+
+	for _, imp := range request.Imp {
+		adapterReq, err := a.processImp(request, imp)
+		if err != nil {
+			errors = append(errors, err)
+		}
+
+		if adapterReq != nil {
+			adapterRequests = append(adapterRequests, adapterReq)
+		}
+	}
+	return adapterRequests, errors
+}
+
+func (a *adapter) makeRequest(request *openrtb2.BidRequest, campaignRequestBody CampaignRequestBody) (*adapters.RequestData, error) {
+	campaignRequestBodyJSON, err := json.Marshal(campaignRequestBody)
+	if err != nil {
+		return nil, err
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json")
+	headers.Add("User-Agent", request.Device.UA)
+	return &adapters.RequestData{
+		Method:  "POST",
+		Uri:     a.endpoint,
+		Body:    campaignRequestBodyJSON,
+		Headers: headers,
+	}, err
+}
+
+func (a *adapter) processImp(request *openrtb2.BidRequest, imp openrtb2.Imp) (*adapters.RequestData, error) {
+	var flippExtParams openrtb_ext.ImpExtFlipp
+	params, _, _, err := jsonparser.Get(imp.Ext, "bidder")
+	if err != nil {
+		return nil, fmt.Errorf("flipp params not found. %v", err)
+	}
+	err = json.Unmarshal(params, &flippExtParams)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to extract flipp params. %v", err)
+	}
+
+	publisherUrl, err := url.Parse(request.Site.Page)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to parse site url. %v", err)
+	}
+	prebidRequest := PrebidRequest{
+		CreativeType:            &flippExtParams.CreativeType,
+		PublisherNameIdentifier: &flippExtParams.PublisherNameIdentifier,
+		RequestID:               &request.ID,
+		Height:                  &imp.Banner.Format[0].H,
```

Comment:
should we check here out of index issue for Format[0]?

Reply:
Have the same question as Format array is recommended but not required in  RTB2.5 documentation